profiler/profiler_plot.py

```diff
@@ -0,0 +1,164 @@
+    for idx, line in enumerate(lines):
+        perf_match = re.match(r"\s+\"(.+)\": (\d+\.\d+)\s*,", line)
+        if perf_match:
+            value = float(perf_match.groups()[1])
+            metric = perf_match.groups()[0]
+            result[metric] = value
+        perf_match = re.match(r"\s+\"(.*)\": (\d+)\s*,", line)
+        if perf_match:
+            value = int(perf_match.groups()[1])
+            metric = perf_match.groups()[0]
+            result[metric] = value
+    return result
+
+
+def extract_tiledb_data(data: ProfileData, metric: str) -> Union[int, float, None]:
+    """Read the tile DB stats from a profile stored data
+    Parse it and extract the expected metric
+    """
+    tiledb_stats = getattr(data, str("tiledb_stats"))
+    lines = tiledb_stats.split("\n")
+    for idx, line in enumerate(lines):
+        perf_match = re.match(rf"\s+\"{metric}\": (\d+\.\d+)\s*,", line)
+        if perf_match:
+            value = float(perf_match.groups()[0])
+            print(f"value for tiledb_stats {metric} = {value}")
+            return value
+        perf_match = re.match(rf"\s+\"{metric}\": (\d+)\s*,", line)
+        if perf_match:
+            value = int(perf_match.groups()[0])
+            return value
+    return None
+
+
+def extract_context_data(data: ProfileData, metric: str) -> Union[int, float, None]:
+    """Read the context data from a profile stored data and return the expected metric"""
+    context = data.context
+    if metric in context.keys():
+        return context[metric]
+    else:
+        raise Exception(f"context does not have the following metric {metric}")
+    return None
+
+
+def create_pandas_df(data: List[ProfileData]) -> pd.DataFrame:
+    """Create panda datat frame for all the runs of a given process
+    Columns are metric names and rows are the runs
+    """
+    table = []
+    # collecting metrics
+    tmp = data[0]
+    metrics = [m for m in dir(tmp) if not m.startswith("__")]
+    tdb_metrics = collect_tiledb_stats(tmp).keys()
+    context_metrics = tmp.context.keys()
+    all_metrics = metrics + list(tdb_metrics) + list(context_metrics)
+
+    # collecting data
+    for run_data in data:
+        row = (
+            [getattr(run_data, metric) for metric in metrics]
+            + [collect_tiledb_stats(run_data)[key] for key in tdb_metrics]
+            + [run_data.context[key] for key in context_metrics]
+        )
+        table.append(row)
+    row_labels = [runData.now for runData in data]
+
+    # Creating dataframe
+    df = pd.DataFrame(table, columns=all_metrics, index=row_labels)
+    return df
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "process",
+        nargs="+",
+        help="The main process name (and optional arguments) to be plotted",
+    )
+    parser.add_argument(
+        "-pd",
+        "--pandas",
+        required=False,
+        help="create pandas dataframe and prints it",
+        action="store_true",
+    )
```

Comment:
Thanks for adding! the underlying method is probably more useful than the CLI option, since I'm assuming if we want to work dataframes we'd do that from a script.  As is, Pandas default column and row truncation will hide most of the output to stdout. But fine to keep this!